### DISCUSSION TITLE
Vercel Deployment Error

### DISCUSSION THREAD

**Discussion by KilianBau** [neuefische/questions#674] [Web] [answered]
Hey, I need some help please. I added a subpage to my code to show details about my objects. Then
I created a pull request but my Vercel deployment failed and gives me this error message:

TypeError: Cannot destructure property 'name' of 'currentCountry' as it is undefined.
--
14:53:21.739 | at DetailCountry (/vercel/path0/.next/server/pages/Subpages/[detailCountry].js:65:13)
14:53:21.739 | at Wc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
14:53:21.740 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
14:53:21.740 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
14:53:21.740 | at $c (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:78:98)
14:53:21.744 | at bd (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:77:404)
14:53:21.745 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:217)
14:53:21.745 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:71:479)
14:53:21.746 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
14:53:21.746 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:481)
14:53:21.747 |  
14:53:21.747 | Error occurred prerendering page "/Subpages/[detailCountry]". Read more: https://nextjs.org/docs/messages/prerender-error
14:53:21.747 | TypeError: Cannot destructure property 'name' of 'currentCountry' as it is undefined.
14:53:21.747 | at DetailCountry (/vercel/path0/.next/server/pages/Subpages/[detailCountry].js:65:13)
14:53:21.748 | at Wc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
14:53:21.748 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
14:53:21.748 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
14:53:21.748 | at $c (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:78:98)
14:53:21.748 | at bd (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:77:404)
14:53:21.748 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:217)
14:53:21.749 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:71:479)
14:53:21.749 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
14:53:21.749 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:481)

Here is my [pull request ](https://github.com/KilianBau/capstone-project/pull/12) and the [Vercel link ](https://vercel.com/kilianbau/capstone-project/2KQmPtMZ795dvSTfNpHswogQfCYU).

I'm happy about any help :)

**Comment by Kheirah** [comment, marked as answer]
The query parameters in the `useRouter` hook are `undefined` when a component is first rendered on the server.

<img width="278" alt="Screenshot 2023-08-17 182332" src="https://github.com/neuefische/questions/assets/22839911/220b15a1-253b-44f3-896e-00e5b5f03208">


This happens because Next.js uses server-side rendering by default, and the query parameters aren't available on the server when the page is initially requested. They only become available on the client side after the JavaScript code has been downloaded and executed.

A common way to handle this situation is to add a conditional check to make sure the query parameter is available before trying to use it. This could look something like this:
```js
//in your DetailCountry component
  const router = useRouter();
  const currentPage = router.query.detailCountry;

  //this is what you need to add
  if (!currentPage) {
    return null;
  }

  const currentCountry = countries.find(
    (country) => country.name === currentPage
  );
```
By adding this conditional check, you can prevent any errors that might occur when trying to use the query parameter before it's available.

PS:
The link to your Vercel account only works for yourself. Nobody has access to it (except if you added someone in which case you'd have to upgrade to Pro and pay 20$ per member).

**Reply by KilianBau** [reply]
Thank you for your help, it works now!
